plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java

```diff
@@ -1088,8 +1101,27 @@ public boolean isPageComplete() {
 
 	public IServer saveServer(IProgressMonitor monitor) throws CoreException {
 		model.updateServer();
+		setOpenshiftMavenProfileActive(model.getDeployProject());
 		return model.saveServer(monitor);
 	}
+	
+	@SuppressWarnings("restriction")
+    private void setOpenshiftMavenProfileActive(IProject newDeployProject)  throws CoreException  {
+	    IFile pom = newDeployProject.getFile(IMavenConstants.POM_FILE_NAME);
+	    if (newDeployProject.hasNature(IMavenConstants.NATURE_ID) && pom != null) {                   
```

Comment:
checking IFile "pom" against being null isn't enough I believe. IContainer#getFile always returns an instance, regardless whether it exists or not. I think that we should also check ```pom.isAccessible()```